test/cpp/end2end/xds_end2end_test.cc

```diff
@@ -1830,6 +1826,41 @@ class XdsEnd2endTest : public ::testing::TestWithParam<TestType> {
     return assignment;
   }
 
+  void SetListenerAndRouteConfiguration(
+      int idx, Listener listener, const RouteConfiguration& route_config) {
```

Comment:
(if counting fault injection) building xDS Listener now impacted by 3 parameters: route config, RDS enablement, and filter config. Since the first two is handled in this method, do we plan to accept filter configs in this method as well?

Reply:
No.  I would expect the caller to make those changes to the Listener and/or RouteConfiguration before passing them into this function.